AdobeStockAsset/Model/OAuth/GetToken.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\AdobeStockAsset\Model\OAuth;
+
+use Magento\Framework\HTTP\Client\CurlFactory;
+use Magento\Framework\Serialize\Serializer\Json;
+
+/**
+ * Class GetToken
+ */
+class GetToken
+{
+    /**
+     * Token URI
+     */
+    private const TOKEN_URI = 'https://ims-na1.adobelogin.com/ims/token';
+
+    /** @var CurlFactory */
+    private $curlFactory;
+
+    /** @var Json */
+    private $json;
+
+    /** @var TokenResponseFactory */
+    private $tokenResponseFactory;
+
+    /**
+     * GetToken constructor.
+     * @param CurlFactory $curlFactory
+     * @param Json $json
+     * @param TokenResponseFactory $tokenResponseFactory
+     */
+    public function __construct(
+        CurlFactory $curlFactory,
+        Json $json,
+        TokenResponseFactory $tokenResponseFactory
+    ) {
+        $this->curlFactory = $curlFactory;
+        $this->json = $json;
+        $this->tokenResponseFactory = $tokenResponseFactory;
+    }
+
+    /**
+     * Get access tokens from Adobe stock IMS
+     *
+     * @param string $apiKey
+     * @param string $privateKey
+     * @param string $code
+     * @return TokenResponse
+     * @throws OAuthException
+     */
+    public function execute(string $apiKey, string $privateKey, string $code): TokenResponse
```

Comment:
Please move this logic to a new public method of `\Magento\AdobeStockClient\Model\Client` and add it to corresponding interface